patches/api/0370-More-Projectile-API.patch

```diff
@@ -179,3 +334,29 @@ index 7051e07b4e456aae0ec9e37808b59e5fa62a4027..225ac312613b9e8f3cf680819f2ebe35
 +    void splash();
 +    // Paper end
  }
+diff --git a/src/main/java/org/bukkit/entity/Trident.java b/src/main/java/org/bukkit/entity/Trident.java
+index c8015ff610e3c1222cb368ea1d8a0c2f3785d9c7..fc3fa6b7176445ad2e0bacdd62476e495be2d686 100644
+--- a/src/main/java/org/bukkit/entity/Trident.java
++++ b/src/main/java/org/bukkit/entity/Trident.java
+@@ -38,5 +38,21 @@ public interface Trident extends AbstractArrow, ThrowableProjectile {
+      * @throws IllegalArgumentException if the loyalty level is lower than 0 or greater than 127
+      */
+     void setLoyaltyLevel(int loyaltyLevel);
++
++    /**
++     * Gets if this trident has dealt damage to an
++     * entity yet or has hit the floor.
++     *
```

Comment:
I'd add a "If neither of these conditions are met, this method returns false" clarification here.